permission_prompt/explainer.md

```diff
@@ -0,0 +1,151 @@
+
+If the user decides to grant the permission, then the fetch continues. If not, it fails.
+
+Once the permission is granted, then the document can make arbitrary fetches to the target origin for its entire lifetime, through media tags, XHRs, etc.
+
+The permission is then persisted. The next document belonging to the same initiator origin that declares its intent to access the same server (perhaps at a different origin, if using a raw IP address) does not trigger a permission prompt. The initial CORS preflight response carries the same ID, and the browser recognizes that the document already has permission to access the server.
+
+## What are the alternatives?
+### Blanket permission prompt
+
+A simple "attacker.com wants to connect to the private network: allow/deny" prompt. This resembles [what is done in iOS today](https://support.apple.com/en-mk/HT211870).
+
+Such a prompt is hard to understand for users. What even is the private network? It also grants too much privilege at once. A single misclick or wrong judgment call (users often click away prompts to get back to what they were doing) means the website is free to scan the local network for attacks and fingerprinting purposes.
+
+### Fetch the device info differently
+One could imagine the name and ID of the device being served some other way, e.g. at a well-known URL: `/.well-known/private-network-access`. It would be fetched first, then the prompt displayed, then the CORS preflight would be sent if permission was granted.
+
+In this model, the API used to trigger the permission prompt could be something else than `fetch()`.
+
+This seems to introduce more complexity for not much benefit. Private Network Access already mandates the use of preflights, two additional headers is the easiest for web developers to implement.
+
+### Identify the target differently
+
+Instead of having the target provide a unique ID, we could use:
+
+* the name: subject to collisions if two similar devices exist on the same network
+* the IP address: subject to collisions and churn with DHCP
+* the MAC address: does not work if the target is not directly reachable on layer 2
+* a public key: complex, would need to design a new protocol to make use of the key later
+* the `-id` and the IP address: in case that multiple devices might accidentally choose the same `-id`.
+
+### Ephemeral permission
+
+Removes the need of an ID and the permission only lasts as long as the document that requested it. This would result in permission fatigue for services such as Plex, where the same target is accessed repeatedly.
+
+Alternatively, the permission only lasts shortly for a certain origin, similarly to the life cycle of a login session.
+
+### Automatically deny similar requests
+
+To avoid the abuse of the permission prompt, we might also want some automatically deny operation for similar requests. Here we propose some alternatives:
```

Comment:
```suggestion
To prevent abuse of the permission prompt, we might also want to automatically deny permission in some cases. Here we propose some alternatives:
```